src/rust/ide/src/controller/searcher.rs

```diff
@@ -208,59 +392,196 @@ mod test {
     use utils::test::traits::*;
 
-    #[test]
-    fn reloading_list() {
-        let mut test    = TestWithLocalPoolExecutor::set_up();
-        let client      = language_server::MockClient::default();
-        let module_path = Path::from_mock_module_name("Test");
+    struct Fixture {
+        searcher : Searcher,
+        entry1   : CompletionSuggestion,
+        entry2   : CompletionSuggestion,
+        entry9   : CompletionSuggestion,
+    }
 
-        let completion_response = language_server::response::Completion {
-            results: vec![1,5,9],
-            current_version: default(),
-        };
-        expect_call!(client.completion(
-            module      = "Test.Test".to_string(),
-            position    = TextLocation::at_document_begin().into(),
-            self_type   = None,
-            return_type = None,
-            tag         = None
-        ) => Ok(completion_response));
-
-        let searcher = Searcher {
-            logger          : default(),
-            data            : default(),
-            notifier        : default(),
-            module          : Rc::new(module_path.qualified_module_name("Test")),
-            position        : Immutable(TextLocation::at_document_begin()),
-            database        : default(),
-            language_server : language_server::Connection::new_mock_rc(client),
-        };
-        let entry1 = model::suggestion_database::Entry {
-            name          : "TestFunction1".to_string(),
-            kind          : model::suggestion_database::EntryKind::Function,
-            module        : "Test.Test".to_string(),
-            arguments     : vec![],
-            return_type   : "Number".to_string(),
-            documentation : default(),
-            self_type     : None
-        };
-        let entry2 = model::suggestion_database::Entry {
-            name : "TestFunction2".to_string(),
-            ..entry1.clone()
-        };
-        searcher.database.put_entry(1,entry1);
-        let entry1 = searcher.database.get(1).unwrap();
-        searcher.database.put_entry(9,entry2);
-        let entry2 = searcher.database.get(9).unwrap();
+    impl Fixture {
+        fn new(client_setup:impl FnOnce(&mut language_server::MockClient)) -> Self {
+            let mut client  = language_server::MockClient::default();
+            let module_path = Path::from_mock_module_name("Test");
+            client_setup(&mut client);
+            let searcher = Searcher {
+                logger          : default(),
+                data            : default(),
+                notifier        : default(),
+                module          : Rc::new(module_path.qualified_module_name("Test")),
+                position        : Immutable(TextLocation::at_document_begin()),
+                database        : default(),
+                language_server : language_server::Connection::new_mock_rc(client),
+                parser          : Parser::new_or_panic(),
```

Comment:
why you dont use `..default()` pattern  ? Can Searcher impl Default?

Reply:
I don't think Searcher can impl Default, because there is no such thing as "Default Language Server connection".